src/treepoem/__init__.py

```diff
@@ -149,12 +149,18 @@ def generate_barcode(
     barcode_type: str,
     data: str | bytes,
     options: dict[str, str | bool] | None = None,
+    template_options: dict[str, str] | None = None,
```

Comment:
I don’t like adding "dictionary of arbitrary values" to the API, when in practice there’s a known set of options. Better to take the arguments exactly.

Also, I’m not convinced there’s really a use case for different scale values, as stretching dimensions different will break barcodes. Perhaps we should have one argument: `scale: int = 2` ?